FreydCategoriesForCAP/gap/CategoryOfRows.gi

```diff
@@ -1272,7 +1272,123 @@ InstallGlobalFunction( INSTALL_FUNCTIONS_FOR_CATEGORY_OF_ROWS,
     end );
     
 end );
-   
+
+##
+## use ExternalHom & CoefficientsOfMorphism to derive Hom-Structure
+##
+AddFinalDerivationBundle( # DistinguishedObjectOfHomomorphismStructure,
+                    [
+                      [ BasisOfExternalHom, 1 ],
+                      [ CoefficientsOfMorphism, 2 ],
+                      [ MultiplyWithElementOfCommutativeRingForMorphisms, 2 ],
+                      [ PreComposeList, 2 ],
+                      [ SumOfMorphisms, 1 ]
+                    ],
+                    [
+                      HomomorphismStructureOnObjects,
+                      HomomorphismStructureOnMorphismsWithGivenObjects,
+                      DistinguishedObjectOfHomomorphismStructure,
+                      InterpretMorphismAsMorphismFromDistinguishedObjectToHomomorphismStructure,
+                      InterpretMorphismFromDistinguishedObjectToHomomorphismStructureAsMorphism
+                    ],
+[
+  DistinguishedObjectOfHomomorphismStructure,
+  function ( cat )
+    local range_cat;
+    
+    range_cat := RangeCategoryOfHomomorphismStructure( cat );
+    
+    return CategoryOfRowsObject( range_cat, 1 );
+    
+  end
+],
+[
+  HomomorphismStructureOnObjects,
+  function ( cat, a, b )
+    local range_cat;
+    
+    range_cat := RangeCategoryOfHomomorphismStructure( cat );
+    
+    return CategoryOfRowsObject( range_cat, Length( BasisOfExternalHom( cat, a, b ) ) );
+    
+  end
+],
+[
+  HomomorphismStructureOnMorphismsWithGivenObjects,
+    
+    #            α
+    #      a --------> a'     hom_source := H(a',b) -------> hom_range := H(a,b')
+    #      |           |
+    #  αℓβ |           | ℓ
+    #      |           |
+    #      v           v
+    #      b' <------- b
+    #            β
+    
+  function ( cat, hom_source, alpha, beta, hom_range )
+    local basis, m;
+    
+    basis := BasisOfExternalHom( cat, Range( alpha ), Source( beta ) );
+    
+    basis := List( basis, ell -> CoefficientsOfMorphism( cat, PreComposeList( cat, [ alpha, ell, beta ] ) ) );
+    
+    m := HomalgMatrix( basis, RankOfObject( hom_source ), RankOfObject( hom_range ), CommutativeRingOfLinearCategory( cat ) );
+    
+    return CategoryOfRowsMorphism( hom_source, m, hom_range );
+    
+  end
+],
+[
+  InterpretMorphismAsMorphismFromDistinguishedObjectToHomomorphismStructure,
+  function ( cat, alpha )
+    local range_cat, coeffs, distinguished_object;
+    
+    range_cat := RangeCategoryOfHomomorphismStructure( cat );
+    
+    coeffs := CoefficientsOfMorphism( cat, alpha );
+    
+    coeffs := HomalgMatrix( coeffs, 1, Length( coeffs ), CommutativeRingOfLinearCategory( cat ) );
+    
+    distinguished_object := DistinguishedObjectOfHomomorphismStructure( cat );
+    
+    return CategoryOfRowsMorphism( distinguished_object, coeffs, CategoryOfRowsObject( range_cat, NrCols( coeffs ) ) );
```

Comment:
```suggestion
    return CategoryOfRowsMorphism( RangeCategoryOfHomomorphismStructure( cat ), distinguished_object, coeffs, CategoryOfRowsObject( range_cat, NrCols( coeffs ) ) );
```

Reply:
Also, please use the WithGiven version to avoid having to construct source and range.